packages/components/for/src/for.tsx

```diff
@@ -0,0 +1,45 @@
+import type { HTMLUIProps, ThemeProps } from "@yamada-ui/core"
+import type { Dict } from "@yamada-ui/utils"
+
+interface ForOptions<T> {
+  /**
+   * The render function to render each item in the array
+   */
+  children: (item: NonNullable<T>, index: number) => React.ReactNode
+  /**
+   * The array to iterate over
+   */
+  each: readonly T[] | T[] | undefined
+  /**
+   * The fallback content to render when the array is empty
+   */
+  fallback?: React.ReactNode
+}
+
+export interface ForProps<T>
+  extends Omit<HTMLUIProps, "children">,
+    ThemeProps<"For">,
+    ForOptions<T> {}
+
+/**
+ * `For` is a component used to loop over an array and render a component for each item.
+ *
+ * @see Docs https://yamada-ui.com/components/other/for
+ */
+export const For = <T extends Dict | number | string | undefined>(
+  props: ForProps<T>,
+) => {
+  const { children, each, fallback } = props
+
+  if (!each || each.length === 0) {
```

Comment:
LGTM. You might also consider using `Array.isArray(each)` to ensure the input is always an array when performing the length check.

Reply:
Thank you for your idea! I thought  the following type definition would have already made sure `each` is an array, but I still added the array checking to give more protection when doing length check!

```js
 /**
 * The array to iterate over
 */
 each: readonly T[] | T[] | undefined
```